interfaces/cython/cantera/composite.py

```diff
@@ -1306,6 +1326,9 @@ def _make_functions():
     def empty_scalar(self):
         return np.empty(self._shape)
 
+    def empty_strings(self):
+        return np.empty(self._shape, dtype='<U50')
```

Comment:
I looked into this a little more, what do you think of `dtype=object`? That uses a generic Python object as the element in the array, and as such has no limits on the string length. My understanding of the tradeoff is that there is additional memory overhead and processing required to use `object` instead of `U` type. However, in my limited testing, since we are over-provisioning the string, the memory use of `U50` is 25x that of `object` (!), and it takes no longer to create the arrays or assign states.
```python
import cantera as ct
import sys
w = ct.Water()
arr = ct.SolutionArray(w, int(5e5))
arr.TP = [300]*int(5e5), [101325]*int(5e5)  # This takes a while
sys.getsizeof(arr.phase_of_matter)
```
gives me about 4MiB for `object` and 95 MiB for `U50`, and roughly the same amount of time to create `arr` for both cases.

Reply:
Oh! I missed your earlier comment about `object`. I didn't see anything on that StackOverflow link about problems with saving, etc. and the test suite passes when using `object`. Is there something specific you're concerned about?

Reply:
I ran a couple of quick tests, and would say that these are the things to consider:

* `dtype=object` is not the default behavior of strings handled by `numpy.ndarray`, i.e. every time something has to be loaded (e.g. using `genfromtxt`) *latin1* encoded fixed column strings have to be detected and a conversion needs to take place. I confirmed that saving to CSV isn't affected.
* the change will affect the handling of `extra` columns, making some changes necessary (meaning that there may be a risk of merge conflicts due to #838)
* it certainly appears to be doable, but any future modification / feature enhancement has to consider the non-default behavior, as I don't think that it's good to do randomly mix *latin1* fixed column arrays and `object`s pointing to strings.

Let me know about your thoughts.

Reply:
I agree with your concern about bifurcating the behavior between `extra` columns and built in columns, and that reading from data sources would require extra processing. I hadn't thought of that. The balance is against how likely it is to truncate a built-in string (that is, a bug in the built-in column), and I think the balance falls in favor of NumPy's default behavior, i.e., I agree with the choice of `U50` and adding a comment to hopefully avoid the possible bug.

The last concern I have is with respect to setting the endian-ness of the string, I think it should be possible to leave that off and let the implementation decide what to use.

Reply:
Ok - just to test, I pushed a version where `object` is used instead of `np.str_`. It's not *too* difficult but doesn't resolve my reservations (I had left the `extra` out though). The endian-ness comment is well taken - it's indeed not necessary (same for lengths), and going with defaults appears to be safe.

Reply:
**PS:** the extra processing for loading of data isn't really an issue, so it's really the `extra` columns that give me reason to pause. Let me know in case you should reconsider.